client/components/data/domain-management/email/index.jsx

```diff
@@ -48,84 +51,125 @@ function getStateFromStores( props ) {
-	},
-
-	componentWillUpdate() {
-		this.loadDomainsAndSitePlans();
-	},
-
-	loadDomainsAndSitePlans() {
-		const selectedSite = this.props.sites.getSelectedSite();
+	}
 
-		if ( this.prevSelectedSite !== selectedSite ) {
-			fetchDomains( selectedSite.ID );
-			this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+	componentWillUpdate( nextProps ) {
+		const { selectedSite: nextSite } = nextProps;
+		const { selectedSite: prevSite } = this.props;
 
-			this.prevSelectedSite = selectedSite;
+		if ( nextSite !== prevSite ) {
+			this.props.loadDomainsAndSitePlans();
 		}
-	},
+	}
 
 	render() {
 		return (
-			<StoreConnection
-				domains={ this.props.domains }
-				googleAppsUsers={ this.props.googleAppsUsers }
-				googleAppsUsersLoaded={ this.props.googleAppsUsersLoaded }
-				component={ this.props.component }
-				stores={ stores }
-				getStateFromStores={ getStateFromStores }
-				products={ this.props.productsList.get() }
-				selectedDomainName={ this.props.selectedDomainName }
-				selectedSite={ this.props.sites.getSelectedSite() }
-				sitePlans={ this.props.sitePlans }
-				context={ this.props.context } />
+			<div>
+				<QueryProducts />
+				<StoreConnection
+					component={ this.props.component }
+					domains={ this.props.domains }
+					googleAppsUsers={ this.props.googleAppsUsers }
+					googleAppsUsersLoaded={ this.props.googleAppsUsersLoaded }
+					stores={ stores }
+					getStateFromStores={ getStateFromStores }
+					products={ this.props.products }
+					selectedDomainName={ this.props.selectedDomainName }
+					selectedSite={ this.props.selectedSite }
+					sitePlans={ this.props.sitePlans }
+					context={ this.props.context }
+				/>
+			</div>
 		);
 	}
-} );
+}
+
+const loadDomainsAndSitePlans = ( fetchSitePlans, selectedSite, sitePlans ) => {
+	if ( ! selectedSite ) {
+		return;
+	}
 
-export default connect(
-	( state, { selectedDomainName, sites } ) => {
-		const googleAppsUsers = selectedDomainName
-			? getByDomain( state, selectedDomainName )
-			: getBySite( state, sites.getSelectedSite().ID );
-
-		return {
-			googleAppsUsers,
-			googleAppsUsersLoaded: isLoaded( state ),
-			sitePlans: getPlansBySite( state, sites.getSelectedSite() )
-		}
-	},
-	( dispatch, { selectedDomainName, sites } ) => {
-		const googleAppsUsersFetcher = selectedDomainName
-			? () => fetchByDomain( selectedDomainName )
-			: () => fetchBySiteId( sites.getSelectedSite().ID );
-
-		return {
-			fetchGoogleAppsUsers: () => dispatch( googleAppsUsersFetcher() ),
-			fetchSitePlans: ( sitePlans, site ) => {
-				if ( shouldFetchSitePlans( sitePlans, site ) ) {
-					dispatch( fetchSitePlans( site.ID ) );
-				}
-			}
-		}
+	if ( ! shouldFetchSitePlans( sitePlans, selectedSite ) ) {
+		return;
 	}
+
+	fetchDomains( selectedSite.ID );
+	fetchSitePlans( selectedSite.ID );
+};
+
+const mapStateToProps = ( state, { selectedDomainName } ) => {
+	const selectedSite = getSelectedSite( state );
```

Comment:
If we update the sitePlans selector to use `getPlansBySiteId`, we should be able to get away with just fetching the selected site id.
```
const selectedSiteId = getSelectedSiteId( state );
```